nsqd/protocol_v2.go

```diff
@@ -313,6 +330,32 @@ func (p *protocolV2) messagePump(client *clientV2, startedChan chan bool) {
 			}
 			msg.Attempts++
 
+			subChannel.StartInFlightTimeout(msg, client.ID, msgTimeout)
+			client.SendingMessage()
+			err = p.SendMessage(client, msg)
+			if err != nil {
+				goto exit
+			}
+			flushed = false
+		case msg := <-zoneMsgChan:
+			if sampleRate > 0 && rand.Int31n(100) > sampleRate {
+				continue
+			}
+			msg.Attempts++
+
+			subChannel.StartInFlightTimeout(msg, client.ID, msgTimeout)
+			client.SendingMessage()
+			err = p.SendMessage(client, msg)
+			if err != nil {
+				goto exit
+			}
+			flushed = false
```

Comment:
I think this duplication could be reduced, if `msg` was declared outside the select, then after the select if not nil the common msg logic runs.